From 3ca557da191c3823f4af18dc6ad89cb6e0768cad Mon Sep 17 00:00:00 2001
From: Patrick Bruenn <p.bruenn@beckhoff.com>
Date: Wed, 14 Jun 2017 15:46:59 +0200
Subject: [PATCH 2/3] imx: cx9020: try pxe boot, if no vmlinuz on mmc

If no vmlinuz is found on mmc, try to boot from pxe.

Signed-off-by: Patrick Bruenn <p.bruenn@beckhoff.com>

Cover-letter:
Enable pxe boot for mx53cx9020
To make kernel development more easy on CX9020, this scripts enables pxe
boot in the default bootscript. pxe boot is then used as a fallback in
case there is no vmlinuz found on mmc.
END
---
 configs/mx53cx9020_defconfig | 1 +
 include/configs/mx53cx9020.h | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/configs/mx53cx9020_defconfig b/configs/mx53cx9020_defconfig
index e0c597cbe2..9609647f0b 100644
--- a/configs/mx53cx9020_defconfig
+++ b/configs/mx53cx9020_defconfig
@@ -18,6 +18,7 @@ CONFIG_CMD_EXT2=y
 CONFIG_CMD_EXT4=y
 CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
+CONFIG_CMD_PXE=y
 CONFIG_OF_CONTROL=y
 CONFIG_FPGA_ALTERA=y
 CONFIG_FPGA_CYCLON2=y
diff --git a/include/configs/mx53cx9020.h b/include/configs/mx53cx9020.h
index b056298d9a..cff1462d00 100644
--- a/include/configs/mx53cx9020.h
+++ b/include/configs/mx53cx9020.h
@@ -67,6 +67,7 @@
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"fdt_addr_r=0x71ff0000\0" \
+	"pxefile_addr_r=0x73000000\0" \
 	"ramdisk_addr_r=0x72000000\0" \
 	"console=ttymxc1,115200\0" \
 	"uenv=/boot/uEnv.txt\0" \
@@ -81,6 +82,7 @@
 		"rootfstype=${mmcrootfstype} " \
 		"${cmdline}\0" \
 	"loadimage=load mmc ${bootpart} ${loadaddr} ${bootdir}/${bootfile}\0" \
+	"loadpxe=dhcp;setenv kernel_addr_r ${loadaddr};pxe get;pxe boot;\0" \
 	"loadrd=load mmc ${bootpart} ${ramdisk_addr_r} ${bootdir}/${rdfile};" \
 		"setenv rdsize ${filesize}\0" \
 	"loadfdt=echo loading ${fdt_path} ...;" \
@@ -130,6 +132,9 @@
 			"echo debug: [${bootargs}] ... ;" \
 			"echo debug: [bootz ${loadaddr} - ${fdt_addr_r}];" \
 			"bootz ${loadaddr} - ${fdt_addr_r}; " \
+		"else " \
+			"echo loading from dhcp ...; " \
+			"run loadpxe; " \
 		"fi;\0"
 
 #define CONFIG_BOOTCOMMAND \
-- 
2.11.0

